services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_base_scheduler.py

```diff
@@ -138,31 +143,50 @@ async def _triage_changed_tasks(
 
 @dataclass(kw_only=True)
 class ScheduledPipelineParams:
-    cluster_id: ClusterID
-    run_metadata: RunMetadataDict
-    mark_for_cancellation: datetime.datetime | None
-    use_on_demand_clusters: bool
-
-    scheduler_task: asyncio.Task | None = None
-    scheduler_waker: asyncio.Event = field(default_factory=asyncio.Event)
+    scheduler_task: asyncio.Task
+    scheduler_waker: asyncio.Event
 
     def wake_up(self) -> None:
         self.scheduler_waker.set()
 
 
 @dataclass
 class BaseCompScheduler(ABC):
-    scheduled_pipelines: dict[
-        tuple[UserID, ProjectID, Iteration], ScheduledPipelineParams
-    ]
     db_engine: Engine
-    wake_up_event: asyncio.Event = field(default_factory=asyncio.Event, init=False)
     rabbitmq_client: RabbitMQClient
     rabbitmq_rpc_client: RabbitMQRPCClient
     settings: ComputationalBackendSettings
     service_runtime_heartbeat_interval: datetime.timedelta
     redis_client: RedisClientSDK
 
+    # NOTE: this is a trick to be able to inheritate from the class
+    _scheduled_pipelines: dict[
+        tuple[UserID, ProjectID, Iteration], ScheduledPipelineParams
```

Comment:
MINOR: I would user a `NamedTuple`

Reply:
yep good point. will think of it for the next iteration